lib/saml/auth_fail_handler.rb

```diff
@@ -17,15 +19,17 @@ def initialize(saml_response, user, session)
       @context = nil
     end
 
-    def known_error?
-      known_errors = [
-        :clicked_deny?,
-        :auth_too_late?,
-        :auth_too_early?
-      ]
+    def error
+      KNOWN_ERRORS.each do |known_error|
+        return known_error if send("#{known_error}?")
+      end
```

Comment:
Nitpick: 

looking at the code for the individual private methods, I am just curious why we don't attempt to do a hash lookup for an error message first, then check to make sure there is only 1 error? 

I think the logic here has a lot of code duplication, that could be simplified and rendered more readable.